src/System.Console/src/System/ConsolePal.Unix.cs

```diff
@@ -902,10 +903,14 @@ public static string Evaluate(string format, params FormatParam[] args)
                                     ~value);
                                 break;
 
-                            // Augment first two parameters by 1
+                            // Some terminfo files make liberal use of %i to mean increment only one argument, not two.
+                            // To make sure we don't break on these files, check if we have one or two variables and increment
+                            // the proper number according to how many are passed.
```

Comment:
I'd suggest phrasing this a bit differently, just to match a tad better to what we're doing in the revised implementation:
Some terminfo files appear to have a fairly liberal interpretation of %i.  The spec states that %i increments the first two arguments, but some uses occur when there's only a single argument.  To make sure we accomodate these files, we increment the values of up to (but not requiring) two arguments.